### PR TITLE
YLP-561 messaging when a team invites a patient

### PR DESCRIPTION
**Pull request by franck-fourel**
- [x] message template
- [x] hydromail integration
- [x] integrate API calls
- [x] integrate Crew
- [x] add tests
- [ ] add tests for Patient invite

### CODE REVIEW

File: api/hydrophoneApi.go [hunk 1]
```diff
@@ -135,6 +135,8 @@ func (a *Api) SetHandlers(prefix string, rtr *mux.Router) {
 	send.Handle("/invite/{userid}", varsHandler(a.SendInvite)).Methods("POST")
 	// POST /confirm/send/team/invite
 	send.Handle("/team/invite", varsHandler(a.SendTeamInvite)).Methods("POST")
+	// POST /confirm/send/team/invite/patient
+	send.Handle("/team/invite/patient", varsHandler(a.SendTeamInvitePatient)).Methods("POST")
```
Comment (ydrisR) [inline api/hydrophoneApi.go]: Why not use the same route for patients and members? `/team/invite` ?

File: api/invite.go [hunk 2]
```diff
@@ -970,6 +970,119 @@ func (a *Api) SendTeamInvite(res http.ResponseWriter, req *http.Request, vars ma
...
+// @Success 200 {object} models.Confirmation "invite details"
+// @Failure 400 {object} status.Status "userId, teamId and isAdmin were not provided or the payload is missing/malformed"
+// @Failure 401 {object} status.Status "Authorization token is missing or does not provide sufficient privileges"
+// @Failure 403 {object} status.Status "Authorization token is invalid"
+// @Failure 409 {object} status.Status "user already has a pending or declined invite OR user is already part of the team"
+// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service"
+// @Failure 500 {object} status.Status "Internal error while processing the invite, detailled error returned in the body"
+// @Router /confirm/send/team/invite/patient [post]
+// @security TidepoolAuth
+func (a *Api) SendTeamInvitePatient(res http.ResponseWriter, req *http.Request, vars map[string]string) {
```
Comment (ydrisR) [inline api/invite.go]: This function is almost a copy of SendTeamInvite. The only difference is the template type/name and the authorization logic.
Can we reuse the same function or at least factorize?

File: api/invite.go [hunk 3]
```diff
@@ -970,6 +970,119 @@ func (a *Api) SendTeamInvite(res http.ResponseWriter, req *http.Request, vars ma
...
+			Role:             "patient",
+			InvitationStatus: "pending",
+		}
+		if invitedUsr != nil {
+			invite.UserId = invitedUsr.UserID
+			member.UserID = invitedUsr.UserID
+			inviteeLanguage = a.getUserLanguage(invite.UserId, res)
+		}
+
+		if _, err := a.perms.AddTeamMember(tokenValue, member); err != nil {
```
Comment (ydrisR) [inline api/invite.go]: This should not work... AddTeamMember works for admin and medical member only, not for patients

**Review by ydrisR — CHANGES_REQUESTED** [review]
Questions about re-use of existing code and api route.
One change required about perms.AddMember

**Comment by franck-fourel** [comment]
the 2 routes to add members and patients have been merged
To do: the patients are not added into a team as the crew client was not yet available. It will be done shortly.

**Review by ydrisR — APPROVED** [review]
Not complete, but ok to merge.

**Pull request merged**